target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4019-rtl30vw.dts

```diff
@@ -0,0 +1,407 @@
+#include <dt-bindings/soc/qcom,tcsr.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "Cell C RTL30VW";
+	compatible = "cellc,rtl30vw", "qcom,ap-dk01.1-c1", "qcom,ap-dk01.2-c1", "qcom,ipq4019";
+
+	aliases {
+		led-boot = &power_blue;
+		led-failsafe = &power_red;
+		led-running = &power_blue;
+		led-upgrade = &power_red;
+	};
+
+	chosen {
+		bootargs-append = "ubi.mtd=ubifs root=/dev/ubiblock0_0 rootfstype=squashfs ro";
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x80000000 0x10000000>;
+	};
+
+	led_spi {
+		compatible = "spi-gpio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		ranges;
+		num-chipselects = <1>;
+
+		gpio-mosi = <&tlmm 2 1>;
+		cs-gpios = <&tlmm 4 1>;
+		gpio-sck = <&tlmm 58 1>;
+
+		led_gpio: led_gpio@0 {
+			compatible = "fairchild,74hc595";
+			reg = <0>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			registers-number = <2>;
+			registers-default = [fe ff];
+			spi-max-frequency = <1000000>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		power_blue: led0 {
+			gpios = <&led_gpio 0 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:power";
+			default-state = "on";
+		};
+
+		power_red: led1 {
+			gpios = <&led_gpio 1 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:red:power";
+			default-state = "off";
+		};
+
+		led6 {
+			gpios = <&led_gpio 6 GPIO_ACTIVE_LOW>;
+			label = "tp28";
+			default-state = "keep";
+		};
+
+		led7 {
+			gpios = <&led_gpio 7 GPIO_ACTIVE_LOW>;
+			label = "tp27";
+			default-state = "keep";
+		};
+
+		led8 {
+			gpios = <&led_gpio 8 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:wlan2g";
+			default-state = "off";
+			linux,default-trigger = "phy0tpt";
+		};
+
+		led9 {
+			gpios = <&led_gpio 9 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:wlan5g";
+			default-state = "off";
+			linux,default-trigger = "phy1tpt";
+		};
+
+		led10 {
+			gpios = <&led_gpio 10 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:wps";
+			default-state = "off";
+		};
+
+		led11 {
+			gpios = <&led_gpio 11 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:voip";
+			default-state = "off";
+		};
+
+		led12 {
+			gpios = <&led_gpio 12 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:s1";
+			default-state = "off";
+		};
+
+		led13 {
+			gpios = <&led_gpio 13 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:s2";
+			default-state = "off";
+		};
+
+		led14 {
+			gpios = <&led_gpio 14 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:s3";
+			default-state = "off";
+		};
+
+		led15 {
+			gpios = <&led_gpio 15 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:blue:s4";
+			default-state = "off";
+		};
+
+		led50 {
+			gpios = <&tlmm 3 GPIO_ACTIVE_HIGH>;
+			label = "rtl30vw:red:signal";
+			default-state = "off";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		button@1 {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&tlmm 5 GPIO_ACTIVE_LOW>;
+		};
+
+		button@2 {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&tlmm 63 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	soc {
```

Comment:
Please keep this ordered by memory space address in increasing order like other devices do.